rfc/system/0001-flyte-execution-tags.md

```diff
@@ -0,0 +1,124 @@
+# Flyte Execution Tags and Metadata
+
+**Authors:**
+
+- @kumare3
+
+## 1 Executive Summary
+
+Flyte currently provides no visual ways of grouping executions and other
+entities with a “project” and “domain”. Usually a project is used by a team of
+engineers and/or researchers and sometimes they may be  experimenting in
+isolation or running multiple co-horts of experiments. Also there are cases in
+which grouping executions by a group might make them easier to find and
+associate them better. This document provides a solution of how we could
+improve the experience of discovering executions within Flyte. It also provides
+motivation of how this feature could further improve other ecosystem projects.
+
+## 2 Motivation
+
+As a User I want to
+ - Group a certain number of executions into an experiment group - for ease of debugging and discovery.
+ - I want to mark certain executions as “blessed” or “released”. This could be done through providing a semantic version after the execution is successful 
+ - I want to group all “sub launchplan” executions with the parent execution.  
+ - External systems could group executions based on some identifiers.  
+ - Users could name their executions without having to worry about the character limits, uniqueness constraints and limited characterset.
+ - Simplify filtering of certain executions
+
+## 3 Proposed Implementation
+
+### Support for tags
+
+We propose to solve the problem of discovery by supporting arbitrary metadata association with an entity. This is similar to conceppt of “tags” as in AWS. The tags are represented as `“key”: “value”` pairs. In kuberenetes this can be represented using “labels” and “annotations”. Labels and annotations are already supported per execution as documented in - [ExecutionCreateRequest](https://docs.flyte.org/projects/flyteidl/en/latest/protos/docs/admin/admin.html#executioncreaterequest) -> [ExecutionSpec](https://docs.flyte.org/projects/flyteidl/en/latest/protos/docs/admin/admin.html#executionspec). Moreover,  every project supports default labels [Project](https://docs.flyte.org/projects/flyteidl/en/latest/protos/docs/admin/admin.html#project). Thus the final execution will have a the union of the project default labels + the user specified labels as it exists today.
+
+Currently the resultant labels are not persisted and are only applied to the
```

Comment:
Kubernetes labels also wouldn't be set when tagging an execution e.g. in the UI that already finished. Such tags that are set post execution should be saved the same way as tags that are set when launching a workflow.